### DISCUSSION TITLE
Get position of drone in ue4 coordinates

### DISCUSSION THREAD

**Discussion by djdylex** [microsoft/AirSim#3672] [Support Q&A]
Hi, i need to plot the flight path of my drone, but using client.simGetVehiclePose().position is frustrating as it gives only relative position from home coordinates. This is not suitable if i want to plot paths with different start positions (and doing manually was painful).

Is there any method that can give me this information, i'm aware this has been asked two years ago but hoping something has been added since then..

If not then i might look at just creating a method myself somehow, or using gps.

**Comment by rajat2004** [comment]
AFAIK, there's no way currently to get the UE4 data itself. Haven't tested this, but the UE4 position data should be available through the `getActorLocation()` method, a wrapper and it's many usages are there in [PawnSimApi.cpp](https://github.com/microsoft/AirSim/blob/8d21f60e3ac99c4bafa664560f9ea9037228c406/Unreal/Plugins/AirSim/Source/PawnSimApi.cpp#L345). A potentially simple way to verify the data without creating a new API could be to add the data in [`getRecordFileLine`](https://github.com/microsoft/AirSim/blob/8d21f60e3ac99c4bafa664560f9ea9037228c406/Unreal/Plugins/AirSim/Source/PawnSimApi.cpp#L578) which will log it when using the Record feature. If the data's correct and API is required, then you can do all the extra stuff like adding the RPC handler and updating the data, etc which is more cumbersome.

I don't think GPS will work since the Start position is the origin for GPS coordinates, and the data will be relative to the start position as well. Might be wrong though
